json/circe/src/main/scala/sttp/client/circe/SttpCirceApi.scala

```diff
@@ -41,8 +41,9 @@ trait SttpCirceApi {
     */
   def asJsonEither[E: Decoder: IsOption, B: Decoder: IsOption]
       : ResponseAs[Either[ResponseException[E, io.circe.Error], B], Any] = {
-    asJson[B].mapLeft { case HttpError(e, code) =>
-      deserializeJson[E].apply(e).fold(DeserializationException(e, _), HttpError(_, code))
+    asJson[B].mapLeft {
+      case HttpError(e, code)                  => deserializeJson[E].apply(e).fold(DeserializationException(e, _), HttpError(_, code))
+      case de @ DeserializationException(_, _) => de
```

Comment:
```suggestion
      case de: DeserializationException => de
```

Reply:
Unfortunately no, it won't be inferred correctly. It would be inferred as 
`ttp.client.ResponseAs[Either[sttp.client.ResponseException[Any,io.circe.Error],B],Any]` instead of `sttp.client.ResponseAs[Either[sttp.client.ResponseException[E,io.circe.Error],B],Any]`
that's why I've chosen to unapply here for readability

Reply:
Ok, thanks for the explanation.